context/state.tsx

```diff
@@ -376,10 +298,33 @@ function init(): tezosState {
   return emptyState();
 }
 
-let AppStateContext: Context<tezosState | null> =
-  createContext<tezosState | null>(null);
-let AppDispatchContext: Context<Dispatch<action> | null> =
+const AppStateContext = createContext<{
+  state: tezosState;
+  dispatch: React.Dispatch<action>;
+}>({ state: emptyState(), dispatch: () => {} });
+const AppDispatchContext: Context<Dispatch<action> | null> =
   createContext<Dispatch<action> | null>(null);
+
+const AppStateProvider = ({ children }: { children: React.ReactNode }) => {
+  const {
+    state: { userAddress },
+  } = useWallet();
+  const [state, dispatch]: [tezosState, React.Dispatch<action>] = useReducer(
+    (state: tezosState, action: action) =>
+      reducer(state, action, { userAddress: userAddress || "" }),
+    emptyState()
+  );
+
+  return (
+    <AppStateContext.Provider value={{ state, dispatch }}>
+      {children}
+    </AppStateContext.Provider>
+  );
+};
+
+const useAppState = () => useContext(AppStateContext).state;
```

Comment:
In the old code I see that state is often an optional and accessed by `state?.field`. So we assume that the state is always initialized in your version?

Reply:
Yes because I call `init` function when I create the context. It's his initial value.